src/js/index.js

```diff
@@ -880,11 +893,15 @@ function loadWaveform(){
         // For min zoom set overflow to visible so that handles show correctly
         document.querySelector("wave").style.overflow = "visible";
         // Add active region from track start to track end
+        var barscolor = theme.getPropertyValue("--barscolor-color");
```

Comment:
Because this is called only when the waveform is initialized, the color of the bars does not get updated when switching themes. We could perhaps target the `.wavesurfer-handle` class inside `themeInit()` and set the backgroundColor property as `!important` to override the element styling.